libressl: fix package & new versions

aumuell opened the pull request:
replace undefined spec with self.spec, also add new stable versions

aumuell committed:
libressl: fix use of undefined spec

installing this package would trigger this python error:
==> Error: NameError: name 'spec' is not defined

aumuell committed:
libressl: checksum 3.6.3 and 3.7.2

newest stable versions

alecbcs commented on var/spack/repos/builtin/packages/libressl/package.py:
```diff
@@ -16,14 +16,16 @@ class Libressl(AutotoolsPackage):
         args = [
-            "--enable-shared" if "+shared" in spec else "--disable-shared",
-            "--enable-static" if "+static" in spec else "--disable-static",
+            "--enable-shared" if "+shared" in self.spec else "--disable-shared",
+            "--enable-static" if "+static" in self.spec else "--disable-static",
```
Super small nitpick, but it might clean up the package a bit to use `self.enable_or_disable()` similar to [this package](https://github.com/spack/spack/blob/4fd7fa5fc1ddcb994348718e8e5cc3ba80ff8773/var/spack/repos/builtin/packages/dlb/package.py#L42C1-L42C1).

aumuell committed:
libressl: use more convenient enable_or_disable

instead of manually crafting configure arguments, thanks to @alecbcs for
the suggestion
